docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,251 @@ Note that:
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` aspects as below are relevant:
+
+* Associating configurations with cloud's child-device twin
+  * an child-device declares its supported config-types to the plugin via MQTT, containing a child-id as reference to the child-device twin
+* Consuming/Providing files from/to thin-edge via network
+  * any configuration file exchange to/from the external child device is handled by an HTTP based file transfer service provided by thin-edge
+
+## Managing Supported Configuration List of child devices
+
+* The `c8y_configuration_plugin` stores and reads the list of supported configuration files for each child-device from a TOML file; similar to whats described in section [Configuration](#configuration) above. Thereby for each child-device an individual TOML file is used.
+* A child-device can also announce its supported configuration files itself to the `c8y_configuration_plugin` via MQTT. The `c8y_configuration_plugin` uses the information received via MQTT to store it into the child-devices individual TOML file.
+
+Each child-devices configuration TOML file is expected in a subfolder, where the subfolder name is treated as `childid`.
+
+Example:
+
+```
+$ tree /etc/tedge/c8y/
+/etc/tedge/c8y/
+├── c8y-configuration-plugin.toml
+├── child1
+│   └── c8y-configuration-plugin.toml
+└── child2
+    └── c8y-configuration-plugin.toml
+```
+
+In that example the plugin serves configuration management for the thin-edge device it-self, for a child-device with childid `child1` and for a child-device with childid `child2`. 
+
+Similar to section [Configuration](#configuration) above, each child-device configuration file is defined by a record with:
+
+* The full `path` to the file. That path is treated as path pointing into the remote filesystem of the child-device.
+* An optional configuration `type`. If not provided, the `path` is used as `type`.
+* The fields `user`, `group` and octal `mode` (as introduced in section [Configuration](#configuration) above) are not supported for child-devices.  
+
+The `c8y_configuration_plugin` make use of the HTTP file transfer feature of the `tedge_agent` to consume/provide the configuration-file from/to the child-device's remote filesystem `path` (see [section below](#details-to-aspect-2-filetransfer-fromto-external-device) for more details about HTTP file transfer).
+
+The `c8y_configuration_plugin` adds per child-device the file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` implicitely to the child-devices configuration file list. So the list can always be configured from the cloud. The `type` for this self configuration file is `c8y-configuration-plugin`.
```

Comment:
```suggestion
The `c8y_configuration_plugin` implicitly creates and maintains a separate `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` file for each connected child device, with its supported configuration list.
This file is also listed as a supported configuration file of the child device in the cloud,
so that the supported configuration list for the child device can always be configured from the cloud.
The implicit `type` for this file is `c8y-configuration-plugin`.
```